src/Servers/Kestrel/Core/src/Internal/Http2/Http2Stream.cs

```diff
@@ -318,8 +318,8 @@ private bool TryValidatePath(ReadOnlySpan<char> pathSegment)
             try
             {
                 // The decoder operates only on raw bytes
-                var pathBuffer = new byte[pathSegment.Length].AsSpan();
-                for (int i = 0; i < pathSegment.Length; i++)
+                Span<byte> pathBuffer = stackalloc byte[pathSegment.Length];
```

Comment:
In general: it may be better to allocate to a constant size (and potentially slice to the correct size if needed), as the JIT can generate better code therefore.
This is especially true, if "skiplocalsinit = true", because then it's a just a (stack-) pointer move. For "skiplocalsinit = false" it's the pointer move + zeroing, but the logic to do so is still straight forward.

Reply:
The path segment could be any length, and changing this to process the path in chunks would be a non-trival change that I don't want to do right now..

Reply:
Done